spark/src/test/java/org/apache/iceberg/actions/TestRewriteDataFilesAction.java

```diff
@@ -311,6 +324,64 @@ public void testRewriteLargeTableHasResiduals() {
     Assert.assertEquals("Rows must match", records, actualRecords);
   }
 
+  @Test
```

Comment:
I decided to add the serde tests of BaseCombinedScanTask here, because it seems to require non-trivial efforts to create BasedCombinedScanTask by hand. I can move out if we'd like to have separate suite with manually crafted test objects.